inc/features/blocks/namespace.php

```diff
@@ -190,21 +204,49 @@ function handle_views_request( WP_REST_Request $request ) : WP_REST_Response {
 }
 
 /**
- * Map aggregate bucket data to response format.
+ * Map event type aggregate bucket data to response format.
  *
- * @param array $bucket Elasticsearch aggregate bucket data.
+ * @param array $event_buckets Elasticsearch aggregate buckets data.
  * @return array
  */
-function map_terms_bucket( array $bucket ) : array {
+function map_aggregations( array $event_buckets ) : array {
 	$data = [
-		'id' => absint( $bucket['key'] ),
-		'count' => $bucket['doc_count'],
+		'loads' => 0,
+		'views' => 0,
+		'conversions' => 0,
+		'audiences' => [],
 	];
 
-	if ( isset( $bucket['audiences']['buckets'] ) ) {
-		$data['audiences'] = map_terms_bucket( $bucket['audiences']['buckets'] );
+	// Map collected event names to output schema.
+	$event_type_map = [
+		'experienceLoad' => 'loads',
+		'experienceView' => 'views',
+		'conversion' => 'conversions',
+	];
+
+	foreach ( $event_buckets as $event_bucket ) {
+
+		$key = $event_type_map[ $event_bucket['key'] ];
+
+		// Set the total.
+		$data[ $key ] = $event_bucket['doc_count'];
+
+		foreach ( $event_bucket['audiences']['buckets'] as $audience_bucket ) {
+			if ( ! isset( $data['audiences'][ $audience_bucket['key'] ] ) ) {
+				$data['audiences'][ $audience_bucket['key'] ] = [
+					'id' => absint( $audience_bucket['key'] ),
+					'loads' => 0,
+					'views' => 0,
+					'conversions' => 0,
+				];
+			}
+			$data['audiences'][ $audience_bucket['key'] ][ $key ] += $audience_bucket['doc_count'];
+		}
 	}
 
+	// Ensure audiences key is an array.
```

Comment:
This is specifically for the REST API right? Seems like that should be at the API level if so.

Reply:
Yes-ish but I'm not sure how valuable it is to get different data back from `get_views()` as you get from the API. We could change it but seems odd that you'd maybe handle the data differently in PHP compared to JS.